flocker/node/agents/functional/test_gce.py

```diff
@@ -175,3 +191,31 @@ def test_attach_elsewhere_attached_volume(self):
             blockdevice_id=attached_volume.blockdevice_id,
             attach_to=api.compute_instance_id(),
         )
+
+    def test_list_volumes_walks_pages(self):
```

Comment:
I thought I'd seen this test before, but I realise the last time was a test for listing nodes rather than volumes:
 * https://github.com/ClusterHQ/flocker/pull/2658#discussion_r54276493